examples/auth/envoy_config.json

```diff
@@ -443,6 +443,62 @@
                               }
                             }
                           },
+                          {
+                            "decorator": {
+                              "operation": "ingress UnknownOperationName"
```

Comment:
can we do 
ingress UnknownHttpMethodForPath_shelves

Reply:
I don't agree with this. One tracing best practice is to **not** have paths in the trace name, it will make querying difficult. Especially if there is a variable template:

```
ingress UnknownHttpMethodForPath_books/3/
ingress UnknownHttpMethodForPath_books/42/
etc.
```

`UnknownOperationName` should be OK, no need to differentiate between 404 and 405 in the trace. Remember:
- Traces are used to tell latency. There will be no latency difference between 404 and 405, so this adds little value.
- Spans have attribute labels that contain HTTP status code and path. No need to include them in the span name, its duplicate info.

Reply:
Good point!

Reply:
Reference that explains my first point better: https://docs.wavefront.com/tracing_best_practices.html#span-names-best-practices

> A poor choice for the span name is getShoppingMenu_123. If you incorporate the menu ID detail directly into the span names, you might end up with unique span names like getShoppingMenu_122, getShoppingMenu_123, getShoppingMenu_124, … getShoppingMenu_nnn, which all represent calls to the same piece of code. The result is a cardinality explosion!

Reply:
Done

Reply:
Actually I don't fully agree that.
- `Traces are used to tell latency` only. Here it can help provide debug information. 
- It is not duplicate info. For the regex, the trace space doesn't have the matched url template in the labels.

The example doesn't fit our case. The example is more about adding dynamic data from request into the span name while we are adding the url templates from service config. There won't be cardinality explosion since at most, the amount of span is just doubled.

Reply:
Good catch, the cardinality problem does not exist.

I still feel it's unnecessary. I don't see how adding the uri template to a span will help debug a 405. 405 means the method is wrong, the path is just extra noise.

I'm OK with keeping it if you think it's helpful. But please add some logic (and test) to truncate the string to 128 characters. Stackdriver will reject spans with names over 128 characters. This is another reason why I would prefer to keep the names simple if possible.

Reply:
Done. 

Not sure if OP config has restriction on the `ShortName`s(the normal routes' span names) and if we also need to add the length check for them. Add a TODO so far.